### PR TITLE
Safety for radarless Honda

### PR DESCRIPTION
**Pull request by sshane**
Moved from https://github.com/commaai/panda/pull/815 so it's on the main repo

### CODE REVIEW

**Comment by sshane** [comment]
Needs opendbc PR merged first and openpilot bumped

File: board/safety/safety_honda.h [hunk 1]
```diff
@@ -48,9 +49,19 @@ AddrCheckStruct honda_bosch_addr_checks[] = {
...
 #define HONDA_BOSCH_ADDR_CHECKS_LEN (sizeof(honda_bosch_addr_checks) / sizeof(honda_bosch_addr_checks[0]))
 
+// Bosch radarless has pt on bus 0
+AddrCheckStruct honda_radarless_addr_checks[] = {
+  {.msg = {{0x296, 0, 4, .check_checksum = true, .max_counter = 3U, .expected_timestep = 40000U}, { 0 }, { 0 }}},  // SCM_BUTTONS
+  {.msg = {{0x158, 0, 8, .check_checksum = true, .max_counter = 3U, .expected_timestep = 10000U}, { 0 }, { 0 }}},  // ENGINE_DATA
+  {.msg = {{0x17C, 0, 8, .check_checksum = true, .max_counter = 3U, .expected_timestep = 10000U}, { 0 }, { 0 }}},  // POWERTRAIN_DATA
+  {.msg = {{0x326, 0, 8, .check_checksum = true, .max_counter = 3U, .expected_timestep = 100000U}, { 0 }, { 0 }}}, // SCM_FEEDBACK
+};
+#define HONDA_RADARLESS_ADDR_CHECKS_LEN (sizeof(honda_radarless_addr_checks) / sizeof(honda_radarless_addr_checks[0]))
```
Comment (adeebshihadeh) [inline board/safety/safety_honda.h:59]: looks same as `honda_nidec_addr_checks`

File: board/safety/safety_honda.h [hunk 2]
```diff
@@ -262,7 +276,8 @@ static int honda_tx_hook(CANPacket_t *to_send, bool longitudinal_allowed) {
     pedal_pressed = pedal_pressed || gas_pressed_prev;
   }
   bool current_controls_allowed = controls_allowed && !(pedal_pressed);
-  int bus_pt = (honda_hw == HONDA_BOSCH) ? 1 : 0;
+  int bus_pt = ((honda_hw == HONDA_BOSCH) && !honda_bosch_radarless) ? 1 : 0;
```
Comment (adeebshihadeh) [inline board/safety/safety_honda.h:279]: move to a helper

File: tests/safety_replay/test_safety_replay.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
   ("5ab784f361e19b78_2020-06-08--16-30-41.bz2", Panda.SAFETY_SUBARU_LEGACY, 0),        # SUBARU.OUTBACK
   ("bb50caf5f0945ab1|2021-06-19--17-20-18.bz2", Panda.SAFETY_TESLA, 0),                # TESLA.AP2_MODELS
   ("bd6a637565e91581_2021-10-29--22-18-31--1--rlog.bz2", Panda.SAFETY_MAZDA, 0),       # MAZDA.CX9_2021
+  ("f44aa96ace22f34a|2021-12-22--06-22-31.bz2", Panda.SAFETY_HONDA_BOSCH, 8),          # HONDA.CIVIC_2022
```
Comment (adeebshihadeh) [inline tests/safety_replay/test_safety_replay.py:26]: use the flag from the Panda class

File: board/safety/safety_honda.h [hunk 4]
```diff
@@ -212,12 +219,12 @@ static int honda_rx_hook(CANPacket_t *to_push) {
 
     bool stock_ecu_detected = false;
     int bus_rdr_car = (honda_hw == HONDA_BOSCH) ? 0 : 2;  // radar bus, car side
```
Comment (sshane) [inline board/safety/safety_honda.h:221]: Is this correct? "Radar" side here is 2, but it's checking if 0xE4 (STEERING_CONTROL) is on the destination bus (which comes from bus 2), so we want it to check bus 0 if I'm not mistaken.

File: tests/safety/test_honda.py [hunk 5]
```diff
@@ -428,10 +432,11 @@ def _acc_state_msg(self, main_on):
     self.__class__.cnt_acc_state += 1
     return self.packer.make_can_msg_panda("SCM_BUTTONS", self.PT_BUS, values)
 
-  def _button_msg(self, buttons, main_on=False):
+  def _button_msg(self, buttons, main_on=False, bus=None):
+    bus = self.PT_BUS if bus is None else bus
```
Comment (adeebshihadeh) [inline tests/safety/test_honda.py:436]: why not just the default to self.PT_BUS? also is this even used anywhere?

Comment (adeebshihadeh) [reply tests/safety/test_honda.py:436]: oh looks like you just want to use self.BUTTONS_BUS?

File: tests/safety/test_honda.py [hunk 6]
```diff
@@ -255,10 +256,11 @@ def _acc_state_msg(self, main_on):
     self.__class__.cnt_acc_state += 1
     return self.packer.make_can_msg_panda("SCM_FEEDBACK", self.PT_BUS, values)
 
-  def _button_msg(self, buttons, main_on=False):
+  def _button_msg(self, buttons, main_on=False, bus=None):
+    bus = self.PT_BUS if bus is None else bus
```
Comment (adeebshihadeh) [inline tests/safety/test_honda.py:260]: same here

File: tests/safety/test_honda.py (hunk 5 above)
Comment (sshane) [reply tests/safety/test_honda.py:436]: No, we need to receive buttons on bus 0 and send on bus 2 so can't just use one bus.

Comment (sshane) [reply tests/safety/test_honda.py:436]: Also can't default to self variables

**Pull request merged**